storage/storage.go

```diff
@@ -45,9 +36,68 @@ type SampleAppender interface {
 	NeedsThrottling() bool
 }
 
+// SampleAppender is the interface to append samples to both, local and remote
+// storage. All methods are goroutine-safe.
+type SampleAppender interface {
+	Throttler
+	// Append appends a sample to the underlying storage. Depending on the
+	// storage implementation, there are different guarantees for the fate
+	// of the sample after Append has returned. Remote storage
+	// implementation will simply drop samples if they cannot keep up with
+	// sending samples. Local storage implementations will only drop metrics
+	// upon unrecoverable errors.
+	Append(*model.Sample) error
+}
+
+// BatchingSampleAppender is the interface to append samples in a batch
+// context, such that it is possible to get a consistent view whereby all
+// samples appended in the same batch will appear at the same time.  This does
+// not mean that they will be appended atomically, but rather that it is
+// possible when querying the local storage implementation to filter out
+// samples appended prior to EndBatch().
+type BatchingSampleAppender interface {
```

Comment:
Maybe "BatchedSampleAppender"? I think with the general idea that this is in the end still a `SampleAppender`, just with two additional wrapper methods, we can generally keep the "appender" naming throughout the code base rather than replacing it by "batcher".

To be fair, both are equally generic names. But I think "appender" is quite established and easily recognized by most people working on the code base.